packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -185,3 +191,10 @@ class _CachedImage {
   final ImageStreamCompleter completer;
   final int sizeBytes;
 }
+
+class _PendingImage {
```

Comment:
Would it make things easier to give this class methods `addListener`/`removeListener` that take a listener, hold on to it in `listener` and register to the the completer instead of relying on users of this class to update both fields?